xray/segment.go

```diff
@@ -68,6 +75,40 @@ func BeginSegment(ctx context.Context, name string) (context.Context, *Segment)
 		seg.GetService().Version = seg.ParentSegment.GetConfiguration().ServiceVersion
 	}
 
+	// Sampling strategy fallbacks to default sampling in the case of directly using BeginSegment API without any instrumentation
+	if r == nil && traceHeader == nil {
+		sd := seg.ParentSegment.GetConfiguration().SamplingStrategy.ShouldTrace(&sampling.Request{})
+		seg.Sampled = sd.Sample
+		logger.Debugf("SamplingStrategy decided: %t", seg.Sampled)
+		seg.AddRuleName(sd)
+	}
+
+	// Sampling strategy for http calls
+	if r != nil && traceHeader != nil {
```

Comment:
Making the above change means this can just be an `else` condition, which would be more concise.